File: 2DAM/PSPRO/ProyectoConAiram/AiramAntonioJuan/src/main/java/com/biblioteca/AiramAntonioJuan/Models/Libro.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.biblioteca.AiramAntonioJuan.Models;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import jakarta.persistence.*;
import java.io.Serializable;
import java.util.Set;

/**
 *
 * @author airam_fckw728
 */


@Entity
@Table(name = "libro")
public class Libro implements Serializable{
    private static final long serialVersionUID = 2L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    @JsonIdentityInfo(
            generator = ObjectIdGenerators.PropertyGenerator.class,
            property = "id"
    )
    private Long id;

    @Column(name = "titulo")
    private String titulo;

    @ManyToOne
    @JoinColumn(name = "id_autor")
    private Autor autor;

    @ManyToMany
    @JoinTable(
            name = "categoria_libro",
            joinColumns = @JoinColumn(name = "id_libro"),
            inverseJoinColumns = @JoinColumn(name = "id_categoria")
    )
    private Set<Categoria> categorias;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getTitulo() {
        return titulo;
    }

    public void setTitulo(String titulo) {
        this.titulo = titulo;
    }

    public Autor getAutor() {
        return autor;
    }

    public void setAutor(Autor autor) {
        this.autor = autor;
    }

    public Set<Categoria> getCategorias() {
        return categorias;
    }

    public void setCategorias(Set<Categoria> categorias) {
        this.categorias = categorias;
    }

    public Libro() {
    }

    @Override
    public String toString() {
        return "Libro{" + "id=" + id + ", titulo=" + titulo + '}';
    }
    
    

}

File: 2DAM/AaRepasoRecuperacionesMarzo/ProgramacionMultimedia/EjerciciosRepaso/Ejercicios Pablo/Ej4_PeqApli/src/app/clases/libro/libro.ts
export class Libro {
    
    private id!: number;
    private titulo!: String;
    private autor!: String;
    private editorial!: String;
    private fechaPublicacion!: Date;
    private static sumaId = 1;

    constructor(titulo:String, autor: String, editorial: String, fechaPublicacion:Date){
        this.id = Libro.sumaId;
        this.titulo = titulo;
        this.autor = autor;
        this.editorial = editorial;
        this.fechaPublicacion = fechaPublicacion;
        Libro.sumaId++;
    }

    getId(){
        return this.id;
    }

    getTitulo(){
        return this.titulo;
    }

    getAutor(){
        return this.autor;
    }

    getEditorial(){
        return this.editorial;
    }

    getFechaPublicacion(){
        return this.fechaPublicacion;
    }

}
